jwst/group_scale/group_scale_step.py

```diff
@@ -16,39 +16,39 @@ class GroupScaleStep(Step):
     def process(self, input):
 
         # Open the input data model
-        with datamodels.RampModel(input) as input_model:
-
-            # Try to get values of NFRAMES and FRMDIVSR to see
-            # if we need to do any rescaling
-            nframes = input_model.meta.exposure.nframes
-            frame_divisor = input_model.meta.exposure.frame_divisor
-
-            # If we didn't find NFRAMES, we don't have enough info
-            # to continue. Skip the step.
-            if nframes is None:
-                self.log.warning('NFRAMES value not found')
-                self.log.warning('Step will be skipped')
-                input_model.meta.cal_step.group_scale = 'SKIPPED'
-                return input_model
```

Comment:
The context manager was removed so that these early returns wouldn't return a closed model, right?  Maybe we should return `input_model.copy()` instead of removing the context manager?  Otherwise `input_model` doesn't get closed if we get to `return result` at the end.

Reply:
This is exactly my thought too.

Generally, pipeline steps should not be modifying their inputs at all, which means we should be returning a deep copy of the input here.  And actually copying it early, before metadata is modified.

Reply:
Forgot about that point and agree. Will open-then-copy.